lib/api/multiObjectDelete.js

```
@@ -475,14 +475,10 @@ function multiObjectDelete(authInfo, request, log, callback) {
         return callback(errors.MissingRequestBodyError);
     }
 
-    if (request.headers['content-md5']) {
-        const md5 = crypto.createHash('md5')
-            .update(request.post, 'utf8').digest('base64');
-        if (md5 !== request.headers['content-md5']) {
-            monitoring.promMetrics('DELETE', request.bucketName, 400,
-                'multiObjectDelete');
-            return callback(errors.BadDigest);
-        }
+    const err = validateChecksumsNoChunking(request.headers, request.post);
+    if (err && err.error !== ChecksumError.MissingChecksum) {
+        log.debug(err, { error: errors.BadDigest });
+        return callback(errors.BadDigest);
```

If we always have the same pattern, it would be better to just return the arsenal error in the function, or nothing, and if there is an error, then we call the callback.
The debug log can be moved there too. Should probably be an error one as it makes the whole API fail?

---

Also, according to the doc:
> Directory bucket - The Content-MD5 request header or a additional checksum request header (including x-amz-checksum-crc32, x-amz-checksum-crc32c, x-amz-checksum-sha1, or x-amz-checksum-sha256) is required for all Multi-Object Delete requests.

In the general case I think it's fine to ignore the `MissingChecksum` error, but here, I believe we want to enforce it, so return an error if the header is not set?

---

I see we have
```
    it('should accept request when content-md5 header is missing', done => {
```

in the test, so I guess that was expected, but that looks not standard. Given your implementation it means there is just no need to return the `MissingChecksum` error, we can just return nothing, if it's fine to not validate anything if the header is not present. That should simplify the processing in the APIs...

---

We cannot require the checksum header yet because we don't support the other algorithms today, so this endpoint would break with the newer SDK's if we require it, after we implement the newer algorithms we can return an error on `MissingChecksum`

---

Can we add a ticket number as TODO so we do not forget anything when this will be done later? As this conflicts with that is in the PR message, that will help reviewers understand the context till this is fixed.

---

What is the conflict with the PR message?

I created this ticket [CLDSRV-747](https://scality.atlassian.net/browse/CLDSRV-747) to enable the requirement when we have the other checksums